tests/tvk-oneclick/tvk_oneclick_test.sh

```diff
@@ -0,0 +1,198 @@
+export input_config=tests/tvk-oneclick/input_config
+
+# shellcheck disable=SC1091
+source tools/tvk-oneclick/tvk-oneclick.sh --source-only
+
+
+#install yq
+sudo snap install yq
+sudo cp /snap/bin/yq /bin/
+
+testinstallTVK() {
+  install_tvk 
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test install_tvk, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testconfigure_ui() {
+  rc=0
+  configure_ui 
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test configure_ui, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testcreate_target() {
+  # debug message
+  # shellcheck disable=SC2154
+  echo "env variable = $nfs_server_ip"
+  # shellcheck disable=SC2154
+  sed -i "s/^\(nfs_server\s*=\s*\).*$/\1\'$nfs_server_ip\'/" "$input_config"
+  create_target
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test create_target, Expected 0 got $rc"
+    kubectl get target tvk-target -n default -o yaml
+  fi
+  return $rc
+}
+
+testsample_test(){
+  sample_test
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testsample_test_helm(){
+  sed -i "s/\(backup_way *= *\).*/\1\'Helm_based\'/" "$input_config"
+  sed -i "s/\(bk_plan_name *= *\).*/\1\'trilio-test-helm\'/" "$input_config"
+  sed -i "s/\(backup_name *= *\).*/\1\'trilio-test-helm\'/" "$input_config"
+  sed -i "s/\(restore_name *= *\).*/\1\'trilio-test-helm\'/" "$input_config"
+  sample_test
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testsample_test_namespace(){
+  sed -i "s/\(backup_way *= *\).*/\1\'Namespace_based\'/" "$input_config"
+  sed -i "s/\(bk_plan_name *= *\).*/\1\'trilio-test-namespace\'/" "$input_config"
+  sed -i "s/\(backup_name *= *\).*/\1\'trilio-test-namespace\'/" "$input_config"
+  sed -i "s/\(restore_name *= *\).*/\1\'trilio-test-namespace\'/" "$input_config"
+  sample_test
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+testsample_test_operator(){
+  sed -i "s/\(backup_way *= *\).*/\1\'Operator_based\'/" "$input_config"
+  sed -i "s/\(bk_plan_name *= *\).*/\1\'trilio-test-operator\'/" "$input_config"
+  sed -i "s/\(backup_name *= *\).*/\1\'trilio-test-operator\'/" "$input_config"
+  sed -i "s/\(restore_name *= *\).*/\1\'trilio-test-operator\'/" "$input_config"
+  sample_test
+  rc=$?
+  # shellcheck disable=SC2181
+  if [ $rc != "0" ]; then
+    # shellcheck disable=SC2082
+    echo "Failed - test sample_test, Expected 0 got $rc"
+  fi
+  return $rc
+}
+
+cleanup() {
```

Comment:
Are you creating these resources that you're trying to delete here?

Reply:
Yes